src/transforms/projectionpursuit.jl

```diff
@@ -0,0 +1,201 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ProjectionPursuit(;tol=1e-6, maxiter=100, deg=5, perc=.9, n=100)
+
+The projection pursuit multivariate transform converts any multivariate distribution into
+the standard multivariate Gaussian distribution.
+
+This iterative algorithm repeatedly finds a direction of projection `α` that maximizes a score of
+non-Gaussianity known as the projection index `I(α)`. The samples projected along `α` are then
+transformed with the [`Quantile`](@ref) transform to remove the non-Gaussian structure. The
+other coordinates in the rotated orthonormal basis `Q = [α ...]` are left untouched.
+
+The non-singularity of Q is controlled by assuring that norm(det(Q)) ≥ `tol`. The iterative 
+process terminates whenever the transformed samples are "more Gaussian" than `perc`% of `n`
+randomly generated samples from the standard multivariate Gaussian distribution, or when the 
+number of iterations reaches a maximum `maxiter`.
+
+# Examples
+
+```julia
+ProjectionPursuit()
+ProjectionPursuit(deg=10)
+ProjectionPursuit(perc=.85, n=50)
+ProjectionPursuit(tol=1e-4, maxiter=250, deg=5, perc=.95, n=100)
+```
+
+"""
+
+struct ProjectionPursuit{T} <: StatelessFeatureTransform
+  tol::T
+  maxiter::Int
+  deg::Int
+  perc::T
+  n::Int
+end
+
+ProjectionPursuit(;tol=1e-6, maxiter=100, deg=5, perc=.95, n=100) =
+  ProjectionPursuit{typeof(tol)}(tol, maxiter, deg, perc, n)
+
+isrevertible(::Type{<:ProjectionPursuit}) = true
+
+# transforms a row of random variables into a convex combination 
+# of random variables with values in [-1,1] and standard normal distribution
+function rscore(Z, α)
+  ᾱ = (1/norm(α)) .* α
+  X = Z * ᾱ
+  2 .* cdf.(Normal(), X) .- 1
+end
+
+# projection index of sample along a given direction
+function pindex(transform, Z, α)
+  d = transform.deg
+  X = rscore(Z, α)
+  I = (3/2) * mean(X)^2
+  if d > 1
+    Pⱼ₋₂, Pⱼ₋₁ = ones(size(X)), X
+    for j = 2:d
+      Pⱼ₋₂, Pⱼ₋₁ = 
+        Pⱼ₋₁, (1/j) * ((2j-1) * X .* Pⱼ₋₁ - (j-1) * Pⱼ₋₂)
+      I += ((2j+1)/2) * (mean(Pⱼ₋₁))^2
+    end
+  end
+  I
+end
+
+# j-th element of the canonical basis in ℝᵈ
+basis(d, j) = 1:d .== j
+
+# index for all vectors in the canonical basis
+function pbasis(transform, Z)
+  q = size(Z, 2)
+  [pindex(transform, Z, basis(q, j)) for j in 1:q]
+end
+
+# projection index of the standard multivariate Gaussian
+function gaussquantiles(transform, N, q)
+  n = transform.n
+  p = transform.perc
+  I = Matrix{Float64}(undef, n, q)
+  for i = 1:n
+    I[i,:] = pbasis(transform, randn(N, q))
+  end
+  quantile.(eachcol(I), p)
```

Comment:
```suggestion
  Is = [pbasis(transform, randn(N, q)) for i in 1:n]
  I  = reduce(hcat, Is)
  quantile.(eachrow(I), p)
```

Julia is a column-major language like Fortran and Matlab. This means that it is more efficient to define loops in columns instead of in rows. Can you please double check that the above change produces the same results as before?

Reply:
I keep getting the "UndefVarError: StatelessFeatureTransform not defined" even after rebase the PR, so I haven't been able to test it.

Reply:
@DianaPat I think you are not doing a rebase correctly. Can you maybe start a fresh PR by copying just the files you modified and pasting them into a new branch?

You need to:

1. git checkout master
2. git checkout -b ppmt
3. copy/paste files from your branch ppmt_transform to the new branch ppmt
4. git push
5. open a new PR